### PR TITLE
bump 17.04.0-rc1

### PR DESCRIPTION
**Pull request by vieux**
ping @docker/core-engine-maintainers

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,67 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Networking
+* Check parameter `--ip`, `--ip6` and `--link-local-ip` in `docker network connect` [#30807](https://github.com/docker/docker/pull/30807)
++ Added support for `dns-search` [#30117](https://github.com/docker/docker/pull/30117)
+
+### Runtime
+* Handle paused container when restoring without live-restore set [#31704](https://github.com/docker/docker/pull/31704)
+- do not allow sub second in healthcheck options in Dockerfile [#31177](https://github.com/docker/docker/pull/31177)
+* make `secret update` support name and id prefix [#30856](https://github.com/docker/docker/pull/30856)
+* The fact Use binary frame for websocket attach endpoint [#30460](https://github.com/docker/docker/pull/30460)
```
Comment (albers) [inline CHANGELOG.md]: This sentence is broken.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,67 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Fix linux mount calls not applying propagation type changes [#30416](https://github.com/docker/docker/pull/30416)
+* Fix ExecIds leak on failed `exec -i` [#30340](https://github.com/docker/docker/pull/30340)
+* Prune named but untagged images if `danglingOnly=true` [#30330](https://github.com/docker/docker/pull/30330)
++ Add daemon flag to set `no_new_priv` as default for unprivileged containers [#29984](https://github.com/docker/docker/pull/29984)
++ Add daemon option `--default-shm-size` [#29692](https://github.com/docker/docker/pull/29692)
++ Aupport registry mirror config reload [#29650](https://github.com/docker/docker/pull/29650)
```
Comment (albers) [inline CHANGELOG.md]: Typo "Aupport"

**Comment by thaJeztah** [comment]
if it makes it, we could add https://github.com/docker/docker/pull/31586

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,67 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Runtime
+* Handle paused container when restoring without live-restore set [#31704](https://github.com/docker/docker/pull/31704)
+- do not allow sub second in healthcheck options in Dockerfile [#31177](https://github.com/docker/docker/pull/31177)
+* make `secret update` support name and id prefix [#30856](https://github.com/docker/docker/pull/30856)
+* The fact Use binary frame for websocket attach endpoint [#30460](https://github.com/docker/docker/pull/30460)
+* Fix linux mount calls not applying propagation type changes [#30416](https://github.com/docker/docker/pull/30416)
+* Fix ExecIds leak on failed `exec -i` [#30340](https://github.com/docker/docker/pull/30340)
+* Prune named but untagged images if `danglingOnly=true` [#30330](https://github.com/docker/docker/pull/30330)
++ Add daemon flag to set `no_new_priv` as default for unprivileged containers [#29984](https://github.com/docker/docker/pull/29984)
```
Comment (albers) [inline CHANGELOG.md]: This should probably be rephrased. Actually no deamon option was added, see #31813

**Comment by vieux** [comment]
@thaJeztah added

**Review by mlaventure — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,69 @@ information on the list of deprecated flags and APIs please have a look at
...
++ Support registry mirror config reload [#29650](https://github.com/docker/docker/pull/29650)
+- Ignore the daemon log config when building images [#29552](https://github.com/docker/docker/pull/29552)
+* Move secret name or ID prefix resolving from client to daemon [#29218](https://github.com/docker/docker/pull/29218)
++ Implement optional ring buffer for container logs [#28762](https://github.com/docker/docker/pull/28762)
++ Allow adding rules to `cgroup devices.allow` on container create/run [#22563](https://github.com/docker/docker/pull/22563)
+
+### Swarm Mode
++ Topology-aware scheduling [#30725](https://github.com/docker/docker/pull/30725)
++ Automatic service rollback on failure [#31108](https://github.com/docker/docker/pull/31108)
++ On a manager node, manager APIs are now invoked through a UNIX socket [docker/swarmkit#1828](https://github.com/docker/swarmkit/pull/1828), [docker/swarmkit#1850](https://github.com/docker/swarmkit/pull/1850), [docker/swarmkit#1851](https://github.com/docker/swarmkit/pull/1851)
```
Comment (tonistiigi) [inline CHANGELOG.md]: This is a bit confusing. The manager API (cluster control API) was always through unix socket. I think this means "Worker and manager on the same node are now connected through a UNIX socket" @aaronlehmann

Comment (aaronlehmann) [inline CHANGELOG.md]: Yes, that's a better description. Also, the CA is accessed through a unix socket.

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,69 @@ information on the list of deprecated flags and APIs please have a look at
...
++ Add `--stop-signal` for `service create` and `service update` [#30754](https://github.com/docker/docker/pull/30754)
++ Add `--read-only` for `service create` and `service update` [#30162](https://github.com/docker/docker/pull/30162)
++ Renew the context after communicating with the registry [#31586](https://github.com/docker/docker/pull/31586)
+
+### Windows
+* Wait for OOBE to prevent crashing during host update [#31054](https://github.com/docker/docker/pull/31054)
+* Block pulling Windows images on non-Windows daemons [#29001](https://github.com/docker/docker/pull/29001)
+
+### DEPRECATION
+* email from `docker login` [#26868](https://github.com/docker/docker/pull/26868)
```
Comment (tonistiigi) [inline CHANGELOG.md]: In previous versions this section means "new deprecations", but in this case, this was deprecated in 1.11 and is now completely removed.

Comment (vieux) [inline CHANGELOG.md]: oh right

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,69 @@ information on the list of deprecated flags and APIs please have a look at
...
++ Add `--filter enabled=true` for `docker plugin ls` [#28627](https://github.com/docker/docker/pull/28627)
++ Add `--format` to `docker service ls` [#28199](https://github.com/docker/docker/pull/28199)
++ Add `publish` and `expose` filter for `docker ps --filter` [#27557](https://github.com/docker/docker/pull/27557)
+* Support multiple service IDs on `docker service ps` [#25234](https://github.com/docker/docker/pull/25234)
++ Allow swarm join with `--availability=drain` [#24993](https://github.com/docker/docker/pull/24993)
+
+### Networking
+* Check parameter `--ip`, `--ip6` and `--link-local-ip` in `docker network connect` [#30807](https://github.com/docker/docker/pull/30807)
++ Added support for `dns-search` [#30117](https://github.com/docker/docker/pull/30117)
+
```
Comment (sanimej) [inline CHANGELOG.md]: Added `--verbose` option for `docker network inspect` to show task details from all swarm nodes [#31710](https://github.com/docker/docker/pull/31710)

Comment (aboch) [inline CHANGELOG.md]: - Clear stale datapath encryption states when joining the cluster [docker/libnetwork#1354](https://github.com/docker/libnetwork/pull/1676)
- Ensure iptables initialization only happens once [docker/libnetwork#1354](https://github.com/docker/libnetwork/pull/1676)

**Comment by vieux** [comment]
ping @sanimej @aboch @tonistiigi

**Comment by sanimej** [comment]
LGTM

**Comment by tonistiigi** [comment]
LGTM

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,73 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Suppressing image digest in docker ps [#30848](https://github.com/docker/docker/pull/30848)
+* Hide command options that are related to Windows [#30788](https://github.com/docker/docker/pull/30788)
+* Fix `docker plugin install` prompt to accept "enter" for the "N" default [#30769](https://github.com/docker/docker/pull/30769)
++ Add `truncate` function for Go templates [#30484](https://github.com/docker/docker/pull/30484)
+* Support expanded syntax of ports in `stack deploy` [#30476](https://github.com/docker/docker/pull/30476)
+* Support expanded syntax of mounts in `stack deploy` [#30597](https://github.com/docker/docker/pull/30597) [#31795](https://github.com/docker/docker/pull/31795)
++ Add `--add-host` for docker build [#30383](https://github.com/docker/docker/pull/30383)
++ Add `.CreatedAt` placeholder for `docker network ls --format` [#29900](https://github.com/docker/docker/pull/29900)
+* Update order of `--secret-rm` and `--secret-add` [#29802](https://github.com/docker/docker/pull/29802)
++ Allow ID-based `docker plugin enable/disable/rm/set` [#29222](https://github.com/docker/docker/pull/29222)
```
Comment (anusha-ragunathan) [inline CHANGELOG.md]: Incorrect. This PR only adds a test. The feature is already available in 17.03

**Pull request merged**